src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -693,7 +693,7 @@ private function createExceptionListener($container, $config, $id, $defaultEntry
 
     private function createSwitchUserListener($container, $id, $config, $defaultProvider, $stateless)
     {
-        $userProvider = isset($config['provider']) ? $this->getUserProviderId($config['provider']) : ($defaultProvider ?: $this->getFirstProvider($id, 'switch_user', $providerIds));
+        $userProvider = isset($config['provider']) ? $this->getUserProviderId($config['provider']) : $defaultProvider;
```

Comment:
This line should not be changed, `$providerIds` argument should be added to the method and passed when calling it from `createFirewall` (where the variable is defined)